check_pgactivity

```diff
@@ -6773,6 +6777,63 @@ SELECT
     return ok( $me, \@msg, \@perfdata );
 }
 
+=item B<wal_receiver> (9.6+)
+
+Check replication status on standby's end.
+
+Perfdata returns wal receiver's pid, current lag and timeline.
+
+This service returns OK if wal receiver's status is 'streaming',
+returns OK if status is different but wal receiver is active, and
+return unknown if not connected.
+
+=cut
+
+sub check_wal_receiver {
+    my $me       = 'POSTGRES_WAL_RECEIVER';
+    my $wal_size = hex('ff000000');
+    my @perfdata;
+    my @rs;
+    my @msg;
+    my @hosts;
+    my %args     = %{ $_[0] };
+
+    @hosts = @{ parse_hosts %args };
+
+    is_compat $hosts[0], 'wal_receiver', $PG_VERSION_96 or exit 1;
+
+    my $query  = qq{ SELECT pid, slot_name, status,
+                     received_lsn, received_tli, latest_end_lsn
+                     FROM pg_stat_wal_receiver };
+
+    @rs = @{ query ( $hosts[0], $query ) };
+
+    return unknown ( $me, ['No receiver connected'], \@perfdata )
+        unless scalar @rs;
+
+    my $pid = $rs[0][0];
+    my $status = $rs[0][2];
+    my $recv_tli = $rs[0][4];
+
+    $rs[0][5] =~ m{^([0-9A-F]+)/([0-9A-F]+)$};
+    my $sender_location = ($wal_size * hex($1) ) + hex($2);
```

Comment:
Yep. I also ask myself about the purpose of this specific check, if it is really useful. We already have check_streaming_delta to detect that a slave is not connected anymore.

I'll find out how developers have come to this view, this will possibly help to decide.